test/Microsoft.Azure.Relay.UnitTests/HybridConnectionTestBase.cs

```diff
@@ -4,39 +4,260 @@
 namespace Microsoft.Azure.Relay.UnitTests
 {
     using System;
-    using Xunit.Abstractions;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Net;
+    using System.Net.WebSockets;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Xunit;
 
-    /// <summary>
-    /// This class is used to ensure that inheriting classes have access to a Connection String Builder and logger.
-    /// </summary>
-    public abstract class HybridConnectionTestBase
+    public class HybridConnectionTestBase
     {
-        protected readonly string AuthenticatedEntityPath = "authenticated";
-        protected readonly string UnauthenticatedEntityPath = "unauthenticated";
+        protected readonly string ConnectionString;
 
-        public HybridConnectionTestBase(ITestOutputHelper output)
+        protected enum EndpointTestType
         {
-            this.Logger = new Logger(output);
+            Authenticated,
+            Unauthenticated
+        }
+
+        public static IEnumerable<object> AuthenticationTestPermutations => new object[]
+        {
+            new object[] { EndpointTestType.Authenticated },
+            new object[] { EndpointTestType.Unauthenticated }
```

Comment:
Do these truly need to be object arrays inside the outer object array?  I guess maybe that's how Xunit does it so one can pass multiple params to the test method.